### PR TITLE
[6] Todo Model

### PR DESCRIPTION
**Pull request by pdany1116**
#### Description

Create Todo resource model

### CODE REVIEW

File: spec/models/todo_spec.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+require 'rails_helper'
+
+RSpec.describe Todo, type: :model do
+  describe 'Validations' do
+    let(:create_todo) { Todo.create(title:, completed:, url:, order:) }
+    let(:title) { Faker::Beer.brand }
+    let(:completed) { false }
+    let(:url) { Faker::Internet.url }
+    let(:order) { 1 }
```
Comment (cazacutudor) [inline spec/models/todo_spec.rb]: We should get some help from factory bot, to generate our Todos, https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#using-factories

File: spec/models/todo_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          todo = create_todo
+
+          expect(todo).not_to be_valid
+        end
+      end
+
+      context 'with invalid completed' do
+        let(:completed) { nil }
+
+        it 'is invalid' do
```
Comment (cazacutudor) [inline spec/models/todo_spec.rb]: Please have a look over https://www.betterspecs.org/

File: db/migrate/20220706073350_create_todos.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+class CreateTodos < ActiveRecord::Migration[7.0]
+  def change
+    create_table :todos do |t|
+      t.string :title
+      t.boolean :completed
+      t.string :url
+      t.integer :order
+
+      t.timestamps
```
Comment (cazacutudor) [inline db/migrate/20220706073350_create_todos.rb:11]: :rocket:

**Review by cazacutudor — CHANGES_REQUESTED** [review]

File: spec/models/todo_spec.rb (hunk 1 above)
Comment (pdany1116) [reply spec/models/todo_spec.rb]: b2bab9ff4dda391f562f6220e99324a3354864e6

File: spec/models/todo_spec.rb (hunk 2 above)
Comment (pdany1116) [reply spec/models/todo_spec.rb]: 5e61a700362b4d84f020053484062b47515a5d71

File: spec/models/todo_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    context 'with invalid url' do
+      subject { build(:todo, url: nil) }
+
+      it { is_expected.not_to be_valid }
+    end
+
+    context 'with invalid order' do
+      subject { build(:todo, order: nil) }
+
+      it { is_expected.not_to be_valid }
```
Comment (cazacutudor) [inline spec/models/todo_spec.rb:34]: Nice :1234:

**Review by cazacutudor — APPROVED** [review]

**Pull request merged**